File: jupyter2/ps/typedef/prng/jl/prng.jl
### begin my answer

### begin hidden
mutable struct Prng
    x :: UInt64
end

mk_prng(seed) = Prng(seed)

function nrand48(rg::Prng)
    y = (0x5DEECE66D * rg.x + 0xB) & ((1 << 48) - 1)
    rg.x = y
    return (y >> 17)
end

### end hidden
### end my answer

function main()
    # @assert ...
    rg = mk_prng(112233445566778899)
    @assert nrand48(rg) == 1781099660
    @assert nrand48(rg) == 328479882
    @assert nrand48(rg) == 1084899894
    @assert nrand48(rg) == 1228799231
    @assert nrand48(rg) == 2081468441
    println("OK")
end

main()

File: jupyter2/ps/typedef/prng/go/prng.go
package main
import "fmt"

/** begin my answer */

/** begin hidden */
type Prng struct {
	x uint64
}

func mk_prng(seed uint64) Prng {
	return Prng{x: seed}
}

func (rg * Prng) nrand48() uint64 {
	y := (rg.x * 0x5DEECE66D + 0xB) & ((1 << 48) - 1)
	rg.x = y
	return y >> 17
}
/** end hidden */
/** end my answer */
func main() {
	rg := mk_prng(112233445566778899)
	if !(rg.nrand48() == 1781099660) { panic("wrong") }
	if !(rg.nrand48() ==  328479882) { panic("wrong") }
	if !(rg.nrand48() == 1084899894) { panic("wrong") }
	if !(rg.nrand48() == 1228799231) { panic("wrong") }
	if !(rg.nrand48() == 2081468441) { panic("wrong") }
	fmt.Println("OK")
}
